diff --git a/dist/index.d.ts b/dist/index.d.ts
index c404b1fd25548dcac5a37e0aa862a00c738a042a..f1c09d353342985217f710aad91da312f9cda901 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -71,6 +71,11 @@ type Config = {
      * @default 60
      */
     maxDuration?: number;
+    /**
+     * The maximum duration to wait for an SSE message response in seconds.
+     * @default 10
+     */
+    sseMessageMaxDuration?: number;
     /**
      * If true, enables console logging.
      * @default false
diff --git a/dist/index.js b/dist/index.js
index 11072f724e74ff9e0b4c3bfec04a6fafa1b0a7bd..b1564a94e43c32252f0dfba9701d7c950560bc81 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -204,6 +204,7 @@ function initializeMcpApiHandler(initializeServer, serverOptions = {}, config =
   sseMessageEndpoint: "/message",
   basePath: "",
   maxDuration: 60,
+  sseMessageMaxDuration: 10,
   verboseLogs: false,
   disableSse: false
 }) {
@@ -214,6 +215,7 @@ function initializeMcpApiHandler(initializeServer, serverOptions = {}, config =
     sseEndpoint: explicitSseEndpoint,
     sseMessageEndpoint: explicitSseMessageEndpoint,
     maxDuration,
+    sseMessageMaxDuration,
     verboseLogs,
     disableSse,
     sessionIdGenerator
@@ -650,7 +652,7 @@ function initializeMcpApiHandler(initializeServer, serverOptions = {}, config =
           logger.log(`Published requests:${sessionId}`, serializedRequest);
           timeout = setTimeout(() => chunkUILOG6SR_js.__async(this, null, function* () {
             yield sendResponse(408, "Request timed out");
-          }), 10 * 1e3);
+          }), (sseMessageMaxDuration != null ? sseMessageMaxDuration : 10) * 1e3);
           res.on("close", () => chunkUILOG6SR_js.__async(this, null, function* () {
             if (!hasResponded) {
               hasResponded = true;
diff --git a/dist/index.mjs b/dist/index.mjs
index c1110409315d81bd1d440dc5910ba720c17f3722..7e5c70c8883c2db721467b2f5923be11fdf4c117 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -198,6 +198,7 @@ function initializeMcpApiHandler(initializeServer, serverOptions = {}, config =
   sseMessageEndpoint: "/message",
   basePath: "",
   maxDuration: 60,
+  sseMessageMaxDuration: 10,
   verboseLogs: false,
   disableSse: false
 }) {
@@ -208,6 +209,7 @@ function initializeMcpApiHandler(initializeServer, serverOptions = {}, config =
     sseEndpoint: explicitSseEndpoint,
     sseMessageEndpoint: explicitSseMessageEndpoint,
     maxDuration,
+    sseMessageMaxDuration,
     verboseLogs,
     disableSse,
     sessionIdGenerator
@@ -644,7 +646,7 @@ function initializeMcpApiHandler(initializeServer, serverOptions = {}, config =
           logger.log(`Published requests:${sessionId}`, serializedRequest);
           timeout = setTimeout(() => __async(this, null, function* () {
             yield sendResponse(408, "Request timed out");
-          }), 10 * 1e3);
+          }), (sseMessageMaxDuration != null ? sseMessageMaxDuration : 10) * 1e3);
           res.on("close", () => __async(this, null, function* () {
             if (!hasResponded) {
               hasResponded = true;
